### PR TITLE
Improved vtk file reading using rectilinear mesh

### PR DESCRIPTION
**Pull request by supriyoumich**
These commits are for external file reading with vtk rectilinear mesh. Currently only unstructured mesh can be read through the input file. However, many problems do not require unstructured mesh. Therefore, rectilinear mesh can reduce the file size and improve the memory usage while reading the files. 

If you want to read rectilinear file, include the following line just below "set Load grain structure" in your input parameter file:
set vtk file type = RECTILINEAR

If you want to read unstructured mesh file, include the following line just below "set Load grain structure" in your input parameter file:
set vtk file type = UNSTRUCTURED
The default option for file reading is set as  UNSTRUCTURED. 

Example rectilinear vtk file named "20_grain_2D_RL.vtk" and parameters.prm file have been uploaded in the **applications/graingrowth_dream3d** for testing purpose.
A pdf file named rectilinear_grid_instructions.pdf explaining how to create a rectilinear grid file from dream3d has been uploaded in the same location **applications/graingrowth_dream3d**.

### CODE REVIEW

File: src/inputFileReader/inputFileReader.cc [hunk 1]
```diff
@@ -729,6 +729,8 @@ inputFileReader::declare_parameters(dealii::ParameterHandler    &parameter_handl
     dealii::Patterns::Anything(),
     "The filename (not including the '.vtk' extension) for the file holding "
     "the grain structure to be loaded.");
+    
+  parameter_handler.declare_entry("vtk file type","UNSTRUCTURED",dealii::Patterns::Anything(),"Whether to load a unstructured file for grain structure.");
```
Comment (landinjm) [inline src/inputFileReader/inputFileReader.cc:734]: I think a multiple selection pattern would be better here.
https://www.dealii.org/current/doxygen/deal.II/classPatterns_1_1MultipleSelection.html

**Review by landinjm — COMMENTED** [review]
Is there a need for the instructions pdf? I feel that these instructions ought to be left to the doxygen documentation or something along those lines.

**Comment by david-montiel-t** [comment]
One option would be to have the instructions a READme.md file within the graingrowth_dream3d app. I am still not sure what would make more sense.

**Comment by landinjm** [comment]
Looks like the merge I made introduced a bunch or errors. I'll retry it a little later this week.

**Pull request closed without merging**